test/test_common/environment.cc

```diff
@@ -49,6 +48,30 @@ void TestEnvironment::initializeOptions(int argc, char** argv) {
   argv_ = argv;
 }
 
+bool TestEnvironment::isTestIpVersionOnly(const Network::Address::IpVersion& type) {
+  const char* option = ::getenv("IP_TEST_TYPE");
+  if (option == nullptr) {
+    return false;
+  }
+  if ((type == Network::Address::IpVersion::v4 && strcmp(option, "v4only") == 0) ||
+      (type == Network::Address::IpVersion::v6 && strcmp(option, "v6only") == 0)) {
```

Comment:
Nit: Prefer using C++ `std::string` where possible (and where we don't have some critical performance issue), since this makes code more robust and readable (C's string library functions are scary fragile and insecure unless you are super careful).